src/features/student/modules/Modules.tsx

```diff
@@ -0,0 +1,84 @@
+import { CourseObj, Course } from '@features/student/modules/Course.tsx';
+import Rating from '@features/student/modules/Rating.tsx';
+import { useQuery } from '@tanstack/react-query';
+import { Collapse, Spin } from 'antd';
+import axios from 'axios';
+import { ItemType } from 'rc-collapse/es/interface';
+import { CSSProperties } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+const panelStyle: CSSProperties = {
+    fontWeight: 'bold',
+    border: 'none',
+};
+
+interface ModulesProps {
+    activeFilter: string;
+}
+
+interface Module {
+    id: number;
+    description: string;
+    number: string;
+    isActive: boolean;
+    grade: number;
+    courses: CourseObj[];
+}
+
+const getCourses = (courses: CourseObj[]) => {
+    return (
+        <>
+            {courses.map((course) => (
+                <Course key={course.id} course={course} />
+            ))}
+        </>
+    );
+};
+
+const getModules = async (activeFilter: string): Promise<ItemType[]> => {
+    const { data: modules } = await axios.get<Module[]>(
+        `${import.meta.env.VITE_BACKEND_API_URL}/module/student/${activeFilter}`
+    );
+
+    return modules
+        .filter((module) => module.isActive)
+        .map((module) => {
+            return {
+                key: module.id,
+                label: module.description,
+                children: getCourses(module.courses),
+                extra: <Rating rating={module.grade} />,
+                style: panelStyle,
+            };
+        });
+};
+
+export default function Modules({ activeFilter }: Readonly<ModulesProps>) {
+    const {
+        isLoading: modulesLoading,
+        error: modulesError,
+        data: modules,
+    } = useQuery({
+        queryKey: ['StudentModules' + activeFilter],
```

Comment:
```suggestion
queryKey: ['studentModules', activeFilter],
```
Query keys should use an array element for each "part" of the key. Also we used camelCase for all keys so we should continue that.
https://tanstack.com/query/v5/docs/react/guides/query-keys#array-keys-with-variables